types/geojson/index.d.ts

```diff
@@ -129,27 +129,27 @@ export interface Polygon extends GeoJsonObject {
  * https://tools.ietf.org/html/rfc7946#section-3.1.7
  */
 export interface MultiPolygon extends GeoJsonObject {
-    type: "MultiPolygon";
+    type: 'MultiPolygon';
     coordinates: Position[][][];
 }
 
 /**
  * Geometry Collection
  * https://tools.ietf.org/html/rfc7946#section-3.1.8
  */
-export interface GeometryCollection extends GeoJsonObject {
-    type: "GeometryCollection";
-    geometries: Geometry[];
+export interface GeometryCollection<G extends Geometry | null = Geometry> extends GeoJsonObject {
+    type: 'GeometryCollection';
+    geometries: G[];
```

Comment:
this is the real change in this PR.

my VSCode fixes some inconsistent code styles.

Reply:
While a Feature's `geometry` may be set to `null` ([ref][1]), I don't see anything in the spec that indicates that `null` is permitted as an element in a `GeometryCollection` ([ref][2]). Please disallow this.

The RFC also says that "implementations SHOULD avoid nested GeometryCollections" which seems like something that could be enforced here and tested.

[1]: https://datatracker.ietf.org/doc/html/rfc7946#section-3.2
[2]: https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.8

Reply:
oh, yes. that is a mistake. had it fixed in the new commit.

I also noticed the two "SHOULD" in the RFC. as the linked RFC ["Key words for use in RFCs to Indicate Requirement Levels"](https://datatracker.ietf.org/doc/html/rfc2119#section-3) explains that "SHOULD" means "there may exist valid reasons in particular circumstances". so I thought we probably better allow it?

Reply:
Fair enough. Could you add a test that `null` is invalid in a `GeometryCollection` and another that allows nested `GeometryCollections` with a comment saying exactly what you said above.

Otherwise looks good to me.